tests/unit/components/sl-input-test.js

```diff
@@ -0,0 +1,210 @@
+import ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+
+moduleForComponent( 'sl-input', 'Unit | Component | sl input', {
+    unit: true
+});
+
+
+test( 'Blur action is triggered when input loses focus', function( assert ) {
+    assert.expect( 1 );
+
+    const done = assert.async();
+
+    this.subject({
+        'blur': 'blur',
+        targetObject: {
+            blur() {
+                assert.ok(
+                    'blur was triggered'
+                );
+
+                done();
+            }
+        }
+    });
+
+    this.$( 'input' ).trigger( 'blur' );
+});
+
+test( 'Click to edit input has the correct class', function( assert ) {
+    this.subject({
+        clickToEdit: true
+    });
+
+    assert.ok(
+        this.$( 'input' ).hasClass( 'click-to-edit' )
+    );
+});
+
+test( 'Input can be disabled' , function( assert ) {
+    this.subject({
+        disabled: true
+    });
+
+    assert.ok(
+        this.$( 'input' ).prop( 'disabled' )
+    );
+});
+
+test( 'Help text is displayed', function( assert) {
+    const helpText = 'Testing help text is displayed';
+
+    this.subject({
+        helpText: helpText
+    });
+
+    assert.equal(
+        this.$( '.help-block' ).text().trim(),
+        helpText
+    );
+});
+
+test( 'Label text is displayed', function( assert ) {
+    const labelText = 'Test label text';
+
+    this.subject({
+        label: labelText
+    });
+
+    assert.equal(
+        this.$( '.control-label' ).text().trim(),
+        labelText
+    );
+});
+
+test( 'Optional property displays optional label', function( assert ) {
+    this.subject({
+        label: 'Optional input',
+        optional: true
+    });
+
+    assert.equal(
+        this.$( '.text-info' ).text().trim(),
+        'Optional'
+    );
+});
+
+test( 'Required property displays required label', function( assert ) {
+    this.subject({
+        label: 'Required input',
+        required: true
+    });
+
+    assert.equal(
+        this.$( '.text-danger' ).text().trim(),
+        'Required'
+    );
+});
+
+test( 'Placeholder property sets the placeholder for the input', function( assert ) {
+    const placeholderText = 'placeholder';
+
+    this.subject({
+        placeholder: 'placeholder'
+    });
+
+    assert.equal(
+        this.$( 'input' ).prop( 'placeholder' ),
+        placeholderText
+    );
+});
+
+test( 'Popover is displayed and has the correct title and content', function( assert ) {
```

Comment:
Is the un-displaying of the popover when the element is unfocused tested in any of the other tests (such as the mixin)?  If not, such a tests should exist here.